Apps/LekaApp/Sources/Views/Widgets/CurationButton.swift

```diff
@@ -0,0 +1,71 @@
+// Leka - iOS Monorepo
+// Copyright APF France handicap
+// SPDX-License-Identifier: Apache-2.0
+
+import ContentKit
+import DesignKit
+import LocalizationKit
+import SwiftUI
+import UtilsKit
+
+// MARK: - CurationButton
+
+public struct CurationButton: View {
+    // MARK: Lifecycle
+
+    public init(_ content: CurationItemModel) {
+        switch content.contentType {
+            case .curation:
+                guard let curation = CategoryCuration(id: content.uuid) else {
+                    fatalError("Content \(content.uuid) is labeled as curation but not decoded as such")
```

Comment:
as the code base and the possibilities grow bigger, it will be harder and harder to test everything.

we might not want to have fatalErrors here but instead return an optional or throw (thought i think an optional makes more sense here) and just doesn't display the content if it doesn't work.

the reason is that in activities, it is mandatory to test that all the exercises and assets are working correctly thus having a fatalError makes it perfectly clear.

but when we will edit the editorial content it might be hard to test everything especially in case of nested curations.

we might even want to display an error widget like with do with the assets in dev/test flight mode but just skip the item when in release mode for prevent user from seeing the error.

Reply:
the same applies to all the widgets